src/DynamoCoreWpf/ViewModels/Core/NodeViewModel.cs

```diff
@@ -1190,40 +1195,66 @@ public void UpdateBubbleContent()
         {
             if (DynamoViewModel == null) return;
 
-            bool hasErrorOrWarning = NodeModel.IsInErrorState || NodeModel.State == ElementState.Warning || NodeModel.State == ElementState.PersistentWarning;
+            bool hasErrorOrWarning = NodeModel.IsInErrorState || NodeModel.State == ElementState.Warning;
+
+            // Persistent warnings should not be dismissed even if nodes are not involved in an execution as they can include:
+            // 1. Compile-time warnings in CBNs
+            // 2. Obsolete nodes with warnings
+            // 3. Dummy or unresolved nodes
+            if (NodeModel.State != ElementState.PersistentWarning)
+            {
+                if (!(NodeModel.WasInvolvedInExecution && hasErrorOrWarning)) return;
+            }
+
+            if (!NodeModel.Infos.Any()) return;
 
-            if (!(NodeModel.WasInvolvedInExecution && hasErrorOrWarning)) return;            
             if (ErrorBubble == null) BuildErrorBubble();
 
             if (!WorkspaceViewModel.Errors.Contains(ErrorBubble)) return;
 
             var topLeft = new Point(NodeModel.X, NodeModel.Y);
             var botRight = new Point(NodeModel.X + NodeModel.Width, NodeModel.Y + NodeModel.Height);
 
+            const InfoBubbleViewModel.Direction connectingDirection = InfoBubbleViewModel.Direction.Bottom;
+            var packets = new List<InfoBubbleDataPacket>(NodeModel.Infos.Count);
+            foreach (var info in NodeModel.Infos)
+            {
+                var infoStyle = info.State == ElementState.Error ? InfoBubbleViewModel.Style.Error : InfoBubbleViewModel.Style.Warning;
+                var data = new InfoBubbleDataPacket(infoStyle, topLeft, botRight, info.Message, connectingDirection);
+                packets.Add(data);
+                //ErrorBubble.UpdateContentCommand.Execute(data);
```

Comment:
should we just remove this? I also found that we update bubble content below anyway it seems